wgengine/netstack/netstack.go

```diff
@@ -62,6 +63,45 @@ import (
 
 const debugPackets = false
 
+// maxInFlightConnectionAttempts returns the global number of in-flight
+// connection limits that we allow for a single netstack Impl. Any new
+// forwarded TCP connections that are opened after the limit has been hit are
+// rejected until the number of pending connections drops below the limit
+// again.
+//
+// Each in-flight connection attempt is a new goroutine and an open TCP
+// connection, so we want to ensure that we don't allow an unbounded number of
+// connections.
+func maxInFlightConnectionAttempts() int {
+	if version.IsMobile() {
+		return 1024 // previous global value
+	}
+	switch version.OS() {
+	case "linux":
+		// On the assumption that most subnet routers deployed in
+		// production are running on Linux, we return a higher value.
+		//
+		// TODO(andrew-d): tune this based on the amount of system
+		// memory instead of a fixed limit.
+		return 8192
```

Comment:
Given that we saw a reasonably small deployment hit 1024 on one of their subnet routers, I wonder if this needs to be higher? If you are operating a fleet of subnet routers, I suspect that detecting and investigating an OOMing client is much easier than tracking down intermittent issues with establishing new connections.

Reply:
Agreed for the question of operating a large fleet of subnet routers in a corporate environment, but this also needs to support home users running an exit node on a Raspberry Pi, along with a wide variety of other subnet router situations that are much more resource-constrained. And for such users, an OOM could break connectivity at a critical time, so I'd rather solve this "properly" instead of bumping the limit to something that might be risky.